src/KingSystem/Damage/dmgDamageManagerBase.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <basis/seadTypes.h>
+#include <container/seadBuffer.h>
+#include <heap/seadExpHeap.h>
+#include <prim/seadRuntimeTypeInfo.h>
+
+#include "KingSystem/ActorSystem/actBaseProcLink.h"
+#include "dmgDamageCallback.h"
+#include "dmgDamageInfoManager.h"
+#include "dmgStruct20.h"
+
+namespace ksys::act {
+class Actor;
+class ActorParam__Data;
+}  // namespace ksys::act
+
+namespace ksys::dmg {
+
+// FIXME: Unknown base. This base seems to handle callbacks and messaging, so maybe a shared base?
+class DamageManagerBase_UnknownBase1 {
+public:
+    DamageManagerBase_UnknownBase1(ksys::act::Actor* WeaponActor);
+    virtual ~DamageManagerBase_UnknownBase1(){};
+
+    // Sturct20 for Damage receive/send?
+    Struct20Base* mStruct20_a = NULL;
+    Struct20Base* mStruct20_b = NULL;
+
+    ksys::act::Actor* mActor = NULL;
+
+    sead::Buffer<DamageCallback*> mCallbacks{};
+
+    // Callback status flags?
+    s32 mField_30 = 0;
+    s8 mField_34 = 0;
+};
+
+// FIXME: Unknown base 2. Helper functions maybe? Might also contain some of the fields from
+// DamageManagerBase.
+class DamageManagerBase_UnknownBase2 {
+public:
+    virtual ~DamageManagerBase_UnknownBase2() = default;
+};
+
+class DamageManagerBase : public DamageManagerBase_UnknownBase1,
+                          public DamageManagerBase_UnknownBase2 {
+public:
+    DamageManagerBase(ksys::act::Actor* WeaponActor);
+    virtual ~DamageManagerBase() = default;
+
+    SEAD_RTTI_BASE(DamageManagerBase)
+
+    virtual u32 getDamage();
+    virtual s32 getField48() { return mField_48; };  // m5
+    virtual s32 getMinDmg() { return mMinDmg; };
+    virtual s32 getField50() { return mField_50; };  // m7
+    virtual s32 getField54() { return mField_54; };  // m8
+    virtual bool checkDamageFlags() { return false; };
+    virtual s32 getFlags2() { return mFlags2; };  // m10
+    virtual void addDamageCallback(s32 eventId, DamageCallback* callback);
+    virtual void removeDamageCallback(DamageCallback* callback);
+    virtual f32 m13() { return 0.0f; };
+    virtual bool m14() { return false; };
+    virtual bool applyDamage(s32& life);
+    virtual bool m16() { return false; };
+    virtual void m17(){};
+    virtual s32 getNumCallbacks();
+    virtual bool initCallbacks(sead::ExpHeap* heap);
```

Comment:
```suggestion
    virtual bool initCallbacks(sead::Heap* heap);
```

This is probably a mistake in the IDB, sead::ExpHeap and sead::Heap were separated at some point but most older functions still reference sead::ExpHeap when the more general sead::Heap should have been used instead.